drivers/display/display_st7735r.c

```diff
@@ -0,0 +1,619 @@
+	ret = st7735r_set_mem_area(data, x, y, desc->width, desc->height);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (desc->pitch > desc->width) {
+		write_h = 1U;
+		nbr_of_writes = desc->height;
+	} else {
+		write_h = desc->height;
+		nbr_of_writes = 1U;
+	}
+
+	ret = st7735r_transmit(data, ST7735R_CMD_RAMWR,
+			       (void *) write_data_start,
+			       desc->width * ST7735R_PIXEL_SIZE * write_h);
+	if (ret < 0) {
+		return ret;
+	}
+
+	tx_bufs.buffers = &tx_buf;
+	tx_bufs.count = 1;
+
+	write_data_start += (desc->pitch * ST7735R_PIXEL_SIZE);
+	for (write_cnt = 1U; write_cnt < nbr_of_writes; ++write_cnt) {
+		tx_buf.buf = (void *)write_data_start;
+		tx_buf.len = desc->width * ST7735R_PIXEL_SIZE * write_h;
+		ret = spi_write(data->spi_dev, &data->config->spi_config, &tx_bufs);
+		if (ret < 0) {
+			return ret;
+		}
+		write_data_start += (desc->pitch * ST7735R_PIXEL_SIZE);
+	}
+
+	return 0;
+}
+
+static void *st7735r_get_framebuffer(const struct device *dev)
+{
+	return NULL;
+}
+
+static int st7735r_set_brightness(const struct device *dev,
+				  const uint8_t brightness)
+{
+	return -ENOTSUP;
+}
+
+static int st7735r_set_contrast(const struct device *dev,
+				const uint8_t contrast)
+{
+	return -ENOTSUP;
+}
+
+static void st7735r_get_capabilities(const struct device *dev,
+				     struct display_capabilities *capabilities)
+{
+	struct st7735r_config *config = (struct st7735r_config *)dev->config;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = config->width;
+	capabilities->y_resolution = config->height;
+	if (config->madctl & ST7735R_MADCTL_BGR) {
+		capabilities->supported_pixel_formats = PIXEL_FORMAT_BGR_565;
+		capabilities->current_pixel_format = PIXEL_FORMAT_BGR_565;
+	} else {
+		capabilities->supported_pixel_formats = PIXEL_FORMAT_RGB_565;
+		capabilities->current_pixel_format = PIXEL_FORMAT_RGB_565;
+	}
+	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+}
+
+static int st7735r_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pixel_format)
+{
+	struct st7735r_config *config = (struct st7735r_config *)dev->config;
+
+	if ((pixel_format == PIXEL_FORMAT_RGB_565) &&
+	    (~config->madctl & ST7735R_MADCTL_BGR)) {
+		return 0;
+	}
+	if ((pixel_format == PIXEL_FORMAT_BGR_565) &&
+	    (config->madctl & ST7735R_MADCTL_BGR)) {
+		return 0;
+	}
+
+	LOG_ERR("Pixel format change not implemented");
+	return -ENOTSUP;
+}
+
+static int st7735r_set_orientation(const struct device *dev,
+				   const enum display_orientation orientation)
+{
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+		return 0;
+	}
+	LOG_ERR("Changing display orientation not implemented");
+	return -ENOTSUP;
+}
+
+static int st7735r_lcd_init(struct st7735r_data *data)
+{
+	const struct st7735r_config *config = data->config;
+	int ret;
+
+	st7735r_set_lcd_margins(data, data->x_offset, data->y_offset);
+
+	ret = st7735r_transmit(data, ST7735R_CMD_FRMCTR1, config->frmctr1,
```

Comment:
It may be sufficient to link all transmit results with OR operator and remove all the `if (ret < 0)` here.

Reply:
I've chosen not to change this. 
grep'ing in the source code doesn't indicate that OR'ing is the way to do it